### ISSUE TITLE
Button disabled when assigning null command

### ISSUE THREAD

**Issue by pr8x** [AvaloniaUI/Avalonia#8660]
A normal `Button` (with `Command` set to `null` by default) is clickable. However, when I bind `Command` to it and the resolved command is `null` or the binding fails to resolve, the `Button` is deactivated (`IsEffectivelyEnabled` is set tot `false`). This is different from what UWP and WPF are doing. In both frameworks a button (even with a bound null command) remains clickable and is only deactivated IFF  `ICommand.CanExecute()` returns false.

Not sure if this behavior is intentional, but I would prefer going with the UWP/WPF approach. Especially since a button without a (valid) command can still host a flyout, which become unusable as well.

**Comment by grokys** [comment]
Yeah this was done to help with the MVVM pattern because a common technique there is to have e.g. a menu that binds to a command on the open document

```
<MenuItem Header="Do Something" Command="{Binding Document.DoSomething}"/>
```

When `Document` is null you want to disable this menu item. I always found it annoying that WPF doesn't do this automatically so I added that feature.

Of course this was early in the project; we probably wouldn't deviate so much from WPF/UWP if we were writing this now, but this was done very early in the project...

I guess we can add a property which controls this behavior, though I'm interested in why you'd want to use a button whose command failed to bind?

**Comment by pr8x** [comment]
> I guess we can add a property which controls this behavior, though I'm interested in why you'd want to use a button whose command failed to bind?

Well, this also doesn't work when the command binding produces `null` (when binding to `ICommand Foo => null` for example), which seems to be a valid use case. When the button is clickable without a command bound (after constructing it) it should be possible to revert to that state via bindings IMO.
As for failed bindings: Not sure. Intuitively, a failed binding should just log somewhere and not touch the value at all (i.e. the command would stay `null`).

What if we keep the behavior for `MenuItem` but remove it from `Button`? A `MenuItem` with a null command is pretty much useless, so it makes sense to disable it. A `Button` can still have a flyout (or manual `Click` handlers)

**Comment by grokys** [comment]
I guess we can just change the default behavior of this for 11.0 to match WPF/UWP, and maybe add a property to re-enable this behavior?
